iroh/src/endpoint.rs

```diff
@@ -2183,4 +2262,110 @@ mod tests {
             .unwrap()
             .unwrap();
     }
+
+    #[tokio::test]
+    #[traced_test]
+    async fn test_connect_with_opts_0rtt() -> TestResult {
+        async fn spawn_server() -> Result<Endpoint> {
+            let server = Endpoint::builder()
+                .alpns(vec![TEST_ALPN.to_vec()])
+                .relay_mode(RelayMode::Disabled)
+                .bind()
+                .await?;
+
+            // Gets aborted via the endpoint closing causing an `Err`
+            tokio::spawn({
```

Comment:
would be good to test actually sending/receiving some data on the handshake

Reply:
Er... what do you mean? This test is actually doing that?

When you get a 0-RTT connection, you get out the 0-RTT data by running `accept_uni` or `accept_bi` after calling `Connecting::into_0rtt()`.

Reply:
Or - are you saying I should make it an echo server, so that it doesn't just go through the motion, but also makes sure that the data flows in/out *correctly*?